eland/operations.py

```diff
@@ -529,10 +556,19 @@ def _unpack_metric_aggs(
 
                     # Pull multiple values from 'percentiles' result.
                     if es_agg[0] == "percentiles":
-                        agg_value = agg_value["values"]
-
-                    agg_value = agg_value[es_agg[1]]
+                        agg_value = agg_value["values"]  # Returns dictionary
+                        if pd_agg == "median":
+                            agg_value = agg_value["50.0"]
+                        # Currently Pandas does the same
+                        # If we call quantile it returns the same result as of median.
+                        elif pd_agg == "quantile" and is_dataframe_agg:
+                            agg_value = agg_value["50.0"]
+                        else:
+                            # Maintain order of percentiles
+                            percentile_values = [agg_value[str(i)] for i in percentiles]
```

Comment:
The exactness of `str(i)` versus the very in-exactness of floats worries me, have we tried running quantile on floats that have long mantissas (ie 50.123456789) and seeing what comes out of Elasticsearch? We may need to do some rounding/smarter here?

Reply:
So, this is what happens here
```python
percentiles
[ 30.0, 40.0,20.0]
agg_value
{'20.0': 361.0407676152134, '30.0': 462.4940487330134, '40.0': 553.3208526972128}
```
This is the state of variables.
So, getting these from ES in a different order
I am basically maintaining the order here and just indexing the percentiles 

the output will be
```python
percentile_values
[361.0407676152134, 462.4940487330134, 553.3208526972128]
```
which directly goes into dataframe.

Reply:
Right but what if I do `ed_df.quantile(0.123456789012345678901234567890)`? I'm sure at some point Elasticsearch truncates the value whereas Python might handle the float differently.

Reply:
Python is truncating first of all
```python
>>> x = 12.3456789012345681231123123123
>>> x
12.345678901234567
```
So, this value is accepted by ES as-is and its returning as-is

Let's assume python is giving to ES `12.3456789012345681231123123123`
But ES is truncating as `12.345678901234567`
So, Py and ES are doing similar. 💯 
Example:
```bash
curl -X GET "localhost:9200/flights/_search?pretty" -H 'Content-Type: application/json' -d'
{
    "size": 0,
    "aggs":
    {"percentiles_AvgTicketPrice": {"percentiles": {"field": "AvgTicketPrice", "percents": [12.3456789012345681231123123123]}},
    "percentiles_Cancelled": {"percentiles": {"field": "Cancelled", "percents": [12.3456789012345681231123123123]}},
    "percentiles_DistanceKilometers": {"percentiles": {"field": "DistanceKilometers", "percents": [12.3456789012345681231123123123]}},
    "percentiles_DistanceMiles": {"percentiles": {"field": "DistanceMiles", "percents": [12.3456789012345681231123123123]}},
    "percentiles_FlightDelay": {"percentiles": {"field": "FlightDelay", "percents": [12.3456789012345681231123123123]}},
    "percentiles_FlightDelayMin": {"percentiles": {"field": "FlightDelayMin", "percents": [12.3456789012345681231123123123]}},
    "percentiles_FlightTimeHour": {"percentiles": {"field": "FlightTimeHour", "percents": [12.3456789012345681231123123123]}},
    "percentiles_FlightTimeMin": {"percentiles": {"field": "FlightTimeMin", "percents": [12.3456789012345681231123123123]}},
    "percentiles_dayOfWeek": {"percentiles": {"field": "dayOfWeek", "percents": [12.3456789012345681231123123123]}}
    }
}
'
{
  "took" : 28,
  "timed_out" : false,
  "_shards" : {
    "total" : 1,
    "successful" : 1,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : {
      "value" : 10000,
      "relation" : "gte"
    },
    "max_score" : null,
    "hits" : [ ]
  },
  "aggregations" : {
    "percentiles_Cancelled" : {
      "values" : {
        "12.345678901234567" : 0.0,
        "12.345678901234567_as_string" : "false"
      }
    },
    "percentiles_FlightDelay" : {
      "values" : {
        "12.345678901234567" : 0.0,
        "12.345678901234567_as_string" : "false"
      }
    },
    "percentiles_FlightDelayMin" : {
      "values" : {
        "12.345678901234567" : 0.0
      }
    },
    "percentiles_DistanceKilometers" : {
      "values" : {
        "12.345678901234567" : 839.1099135151572
      }
    },
    "percentiles_FlightTimeMin" : {
      "values" : {
        "12.345678901234567" : 76.20407994639103
      }
    },
    "percentiles_FlightTimeHour" : {
      "values" : {
        "12.345678901234567" : 1.2700679853583805
      }
    },
    "percentiles_DistanceMiles" : {
      "values" : {
        "12.345678901234567" : 521.3987255553205
      }
    },
    "percentiles_dayOfWeek" : {
      "values" : {
        "12.345678901234567" : 0.0
      }
    },
    "percentiles_AvgTicketPrice" : {
      "values" : {
        "12.345678901234567" : 284.3243689582871
      }
    }
  }
}
```

Reply:
Also this output is matching with pandas. 
Do you want me to add a test for this. Just in case ?